src/gpg2john.c

```diff
@@ -836,17 +839,17 @@ string_to_key(void)
 		m_hashAlgorithm = Getc();
 		// hash_algs(Getc());
 		if (gpg_dbg) fprintf(stderr, "\t\tSalt - ");
-		if (gpg_dbg)
-			dump(8);
-		else
-			give(8, m_salt, sizeof(m_salt));
-		if (gpg_dbg) fprintf(stderr, "\n");
-		{
-			int count, c = Getc();
-			count = (16 + (c & 15)) << ((c >> 4) + EXPBIAS);
-			if (gpg_dbg) fprintf(stderr, "\t\tCount - %d(coded count %d)\n", count, c);
-			m_count = count;
+		give(8, m_salt, sizeof(m_salt));
+		if (gpg_dbg) {
+			for (i = 0; i < 8; ++i) {
+				fprintf(stderr, "%02x", m_salt[i]);
+			}
+			fprintf(stderr, "\n");
 		}
+		int count, c = Getc();
+		count = (16 + (c & 15)) << ((c >> 4) + EXPBIAS);
+		if (gpg_dbg) fprintf(stderr, "\t\tCount - %d(coded count %d)\n", count, c);
+		m_count = count;
```

Comment:
Please revert the non-requested changes here. Having the scope of those two variables limited actually made sense.

Reply:
sorry, pushed too fast...